roadmap/state_sync_protocol/state-sync-protocol.md

```diff
@@ -0,0 +1,204 @@
+
+### Memory / Storage Estimate
+
+Access nodes will need to store the full indexed state and the Execution Data for each block on disk. They also need to be able to load a state snapshot / complete trie in memory in order to apply updates.
+
+**Execution state checkpoint (complete)**
+
+```console
+simon_zhu_dapperlabs_com@execution-001 /var/flow/data/execution $ ls -lh | tail -n 7
+-rw------- 1 root root  27G Oct 24 12:18 checkpoint.00010182
+-rw------- 1 root root  27G Oct 24 19:45 checkpoint.00010251
+-rw------- 1 root root  27G Oct 25 03:25 checkpoint.00010315
+-rw------- 1 root root  27G Oct 25 11:01 checkpoint.00010356
+-rw------- 1 root root  27G Oct 25 18:43 checkpoint.00010405
+drwxr-xr-x 2 root root 4.0K Oct  6 19:06 extralogs
+-rw-r--r-- 1 root root  24G Oct  6 19:02 root.checkpoint
+```
+
+**Execution Data (individual)**
+
+```console
+smnzhu@mbp:~$ SORTED_BY_SIZE=$(gsutil du -h gs://flow_public_mainnet13_execution_state | sed -e 's/ MiB/M/g' -e 's/ KiB/K/g' | sort -h | sed -e 's/M/ MiB/g' -e 's/K/ KiB/g')
+smnzhu@mbp:~$ echo $SORTED_BY_SIZE | head -n 1
+3.57 KiB     gs://flow_public_mainnet13_execution_state/5d00b1ef2670d5987015c582727ab2308b19916ea4303ad56d600bb1fb4a12fc.cbor
+smnzhu@mbp:~$ echo $SORTED_BY_SIZE | tail -n 1
+377.18 MiB   gs://flow_public_mainnet13_execution_state/38db1ad86322fa81ff105328fd2812f2d5eed87ea4b7dba4e4095b1934fcec86.cbor
+smnzhu@mbp:~$ echo $SORTED_BY_SIZE | awk '{ a[i++]=$0; } END { print a[int(i/2)]; }'
+70.47 KiB    gs://flow_public_mainnet13_execution_state/4f6892c5b1b16e41ae1adda086b8aa241d254b780f44ceb5e45a5a1d5bb3d8aa.cbor
+```
+
+**Execution Data (~3 week spork)**
+
+```console
+smnzhu@mbp:~$ gsutil du -sh gs://flow_public_mainnet13_execution_state
+446.14 GiB   gs://flow_public_mainnet13_execution_state
+smnzhu@mbp:~$ gsutil ls gs://flow_public_mainnet13_execution_state | wc -l
+  473590
+```
+
+**Execution State Index (~3 week spork)**
+
+```console
+simon_zhu_dapperlabs_com@observer-001 /var/flow/data/mainnet-13 $ sudo du -sh index
+103G    index
+```
+
+Based on the numbers above, we should require Access nodes to have >43GB of RAM (27GB + current 16GB requirement) and minimum 1.5-2TB of storage space for a 3 month spork. These requirements can likely be decreased if sharding is implemented (see [Future Optimizations](#future-optimizations)).
+
+## Future Optimizations
+
+### Checkpointing
+
+Having a checkpointing mechanism will allow new nodes to catch up more quickly, while also reducing the amount of data Access nodes need to store on disk.
+
+Access nodes can periodically (e.g. at every 10000th block) generate a snapshot of the full state index and store it in their blobstore. They can then prune all of the Execution Data for blocks older than the snapshot, and require nodes that are behind to directly sync the snapshot first before catching up via per-block Execution Data updates.
```

Comment:
Node operators might want to keep the historic state; pruning might not be desirable 

```suggestion
Access nodes can periodically (e.g. at every 10000th block) generate a snapshot of the full state index and store it in their blobstore. Thereby, new access nodes (and other external clients) could directly sync starting from the most recent snapshot and catch up via per-block Execution Data updates (before they sync the full state history if they desire to do so).
```